submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/submitter/ProjectSubmitter.kt

```diff
@@ -1,36 +1,45 @@
 package ac.uk.ebi.biostd.submission.submitter
 
 import ac.uk.ebi.biostd.submission.exceptions.InvalidSubmissionException
-import ac.uk.ebi.biostd.submission.processors.IProjectProcessor
+import ac.uk.ebi.biostd.submission.service.ProjectValidationService
 import ac.uk.ebi.biostd.submission.util.AccNoPatternUtil
 import ebi.ac.uk.model.ExtendedSubmission
 import ebi.ac.uk.model.Submission
 import ebi.ac.uk.model.constants.ProcessingStatus.PROCESSED
 import ebi.ac.uk.model.extensions.accNoTemplate
+import ebi.ac.uk.model.extensions.addAccessTag
 import ebi.ac.uk.persistence.PersistenceContext
 import org.springframework.transaction.annotation.Isolation
 import org.springframework.transaction.annotation.Transactional
 
 open class ProjectSubmitter(
     private val accNoPatternUtil: AccNoPatternUtil,
-    private val processors: List<IProjectProcessor>
+    private val persistenceContext: PersistenceContext,
+    private val projectValidationService: ProjectValidationService
 ) {
     @Transactional(isolation = Isolation.READ_UNCOMMITTED)
-    open fun submit(project: ExtendedSubmission, context: PersistenceContext): Submission {
-        val processingErrors = process(project, context)
+    open fun submit(project: ExtendedSubmission): Submission {
+        val processingErrors = process(project)
         if (processingErrors.isEmpty()) {
             val sequencePrefix = accNoPatternUtil.getPattern(project.accNoTemplate!!)
-            context.createAccNoPatternSequence(sequencePrefix)
-            context.saveAccessTag(project.accNo)
+            persistenceContext.createAccNoPatternSequence(sequencePrefix)
+            persistenceContext.saveAccessTag(project.accNo)
             project.processingStatus = PROCESSED
-            return context.saveSubmission(project)
+            return persistenceContext.saveSubmission(project)
         }
 
         throw InvalidSubmissionException("Submission validation errors", processingErrors)
     }
 
-    private fun process(submission: ExtendedSubmission, context: PersistenceContext) =
-        processors
-            .map { runCatching { it.process(submission, context) } }
-            .mapNotNull { it.exceptionOrNull() }
+    private fun process(project: ExtendedSubmission) =
```

Comment:
no need to wrap into a list as single processing result. So simple instance and null check will do the job

Reply:
`IvalidSubmissionException` expects a list of errors in case it fails. I think it's cleaner to leave it like this than having to declare a list later. Also, both submitters would be consistent.

Reply:
do not agree, if the method retrieves a single error, the single error should be retrieved. We should not change it to make looks similar to the submission business case.  I consider consistent for example all variables starting with lower case, or use final keyword for parameter arguments, but not this case. 

About exception, I see from a business perspective that is totally possible that either multiple or single error is thrown, like users having a a single or multiple errors so InvalidSubmissionException should support single error too. 

Said that I totally understand your point, like from code perspective is cleaner retrieve a list and also is simpler to have similar solution for both submission and projects, for me is more important code to be similar to business (if single error is throw single element) but it hard to argument which quality attribute is better. 

In conclusion 1000 lines consider the argument and merge changes (adding more changes or not).There is not clear better solution IMO.

Reply:
![image](https://user-images.githubusercontent.com/19352419/73838010-9c74e200-480a-11ea-9b75-195270130c66.png)
😆